aws/resource_aws_fsx_file_system_test.go

```diff
@@ -0,0 +1,365 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFileSystemExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.0.backup_retention", "3"),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.0.copy_tags_to_backups", "true"),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.0.throughput_capacity", "1024"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"kms_key_id"},
+			},
+		},
+	})
+}
+
+func TestAccAWSFsxFileSystem_windowsUpdate(t *testing.T) {
+	var v fsx.FileSystem
+	resourceName := "aws_fsx_file_system.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: resourceName,
+		Providers:     testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsFsxFileSystemWindowsConfigOpts,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFileSystemExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.0.backup_retention", "3"),
+				),
+			},
+			{
+				Config: testAccAwsFsxFileSystemWindowsUpdateOpts,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFileSystemExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "windows_configuration.0.backup_retention", "30"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckFileSystemExists(n string, v *fsx.FileSystem) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).fsxconn
+
+		request := &fsx.DescribeFileSystemsInput{
+			FileSystemIds: []*string{aws.String(rs.Primary.ID)},
+		}
+
+		response, err := conn.DescribeFileSystems(request)
+		if err == nil {
+			if response.FileSystems != nil && len(response.FileSystems) > 0 {
+				*v = *response.FileSystems[0]
+				return nil
+			}
+		}
+		return fmt.Errorf("Error finding FSx filesystem %s", rs.Primary.ID)
+	}
+}
+
+const testAccAwsFsxFileSystemLustreBasic = `
+resource "aws_vpc" "test" {
+  cidr_block  = "10.0.0.0/16"
+}
+
+resource "aws_subnet" "test" {
+  vpc_id     				= "${aws_vpc.test.id}"
+	cidr_block 				= "10.0.1.0/24"
+	availability_zone = "us-east-1a"
+}
+
+resource "aws_kms_key" "test" {
+  description             = "FSx KMS Testing key"
+  deletion_window_in_days = 7
+}
+
+resource "aws_fsx_file_system" "test" {
+	depends_on 	= ["aws_subnet.test", "aws_kms_key.test"]
```

Comment:
Nit: This `depends_on` is extraneous as the two resources have implicit references below.